public/src/containers/create_campaign.jsx

```diff
@@ -0,0 +1,44 @@
+import { withRouter } from 'react-router-dom';
+import { connect } from 'react-redux';
+import { reduxForm } from 'redux-form';
+
+import CampaignPage from '../components/campaign/campaign';
+
+import { saveNewCampaign } from '../actions/campaign';
+import { fetchAllScripts, fetchAllContactLists } from '../actions/campaign_form';
+
+// TODO: write validation that is applicable to campaign creation!
+// // e.g. check input length, checking uniqueness, etc
+
+function validate(values) {
+  const errors = {};
+  const { password, password_confirm } = values;
+  if (!!password && !!password_confirm && password !== password_confirm) {
+    errors.password_confirm = 'the passwords must match.';
+  }
+  return errors;
+}
+
+function mapStateToProps(state) {
+  return {
+    auth: state.auth,
+    scriptOptions: state.campaign_form.scripts,
+    contactListOptions: state.campaign_form.contacts
```

Comment:
But do you think we should differentiate between contacts and contact_lists? This is displaying contact_lists right? I think saying contacts might be misleading. Or maybe I'm reading it incorrectly?